StorageUtility/Linux/StorageUtility.cpp

```diff
@@ -17,251 +17,181 @@ limitations under the License.
-        {
-            if ( eOnErrorBehavior::Continue == OnErrorBehavior ) { continue; }
-            SetupDiDestroyDeviceInfoList( devInfoHandle );
-            return( eErrorCode::Memory );
-        }
-        pDevDetailData->cbSize = sizeof( SP_INTERFACE_DEVICE_DETAIL_DATA );
+        devicePath = udev_list_entry_get_name(udevListEntry);
+        udevDevice = udev_device_new_from_syspath(udevObject, devicePath);
+        String udeviceNode = udev_device_get_devnode(udevDevice);
 
-        if ( SetupDiGetDeviceInterfaceDetail( devInfoHandle, &devInterfaceData, pDevDetailData, bufferSize, NULL, NULL ) == FALSE )
+        if (true == IsStorageDisk(udevDevice))
         {
-            error = GetLastError();
-            if ( ERROR_INSUFFICIENT_BUFFER != error )
-            {
-                HeapFree(GetProcessHeap(), HEAP_NO_SERIALIZE, pDevDetailData);
-                if ( eOnErrorBehavior::Continue == OnErrorBehavior ) { continue; }
-                SetupDiDestroyDeviceInfoList( devInfoHandle );
-                return( eErrorCode::Unknown );
-            }
+            DevicePaths.push_back(udeviceNode);
+            SysDevicePaths.push_back(devicePath);
         }
 
-        eErrorCode errorCode;
-        errorCode = eErrorCode::Unknown;
-        Callback.Function(Callback.Data, devInfoHandle, devInfoData, devInterfaceData, pDevDetailData, bufferSize, errorCode);
-
-        HeapFree(GetProcessHeap(), HEAP_NO_SERIALIZE, pDevDetailData);
+        udev_device_unref(udevDevice);
     }
+    //! Free the enumerator object
+    //!
+    udev_enumerate_unref(enumerate);
 
-    SetupDiDestroyDeviceInfoList( devInfoHandle );
+    udev_unref(udevObject);
 
-    return( eErrorCode::None );
-}*/
+    return eErrorCode::None;
+}
 
-eErrorCode GetStorageAdapterProperty( HANDLE Handle, sStorageAdapterProperty& AdapterProperty )
+void GetAdapterBus(DeviceHandle& Handle, String SysDevicePath)
 {
-   /* STORAGE_DESCRIPTOR_HEADER storageDescHeader;
-
-    STORAGE_PROPERTY_QUERY StorageProperty;
-    StorageProperty.PropertyId = StorageAdapterProperty;
-    StorageProperty.QueryType = PropertyStandardQuery;
-
-    DWORD bytesReturned;
-    DWORD ret =
-        DeviceIoControl(
-                        Handle,
-                        IOCTL_STORAGE_QUERY_PROPERTY,
-                        &StorageProperty,
-                        sizeof(STORAGE_PROPERTY_QUERY),
-                        &storageDescHeader,
-                        sizeof(STORAGE_DESCRIPTOR_HEADER),
-                        &bytesReturned,
-                        NULL
-                        );
-    if ( 0 == ret )
-    {
-        return( eErrorCode::Io );
-    }
+    struct udev *udevObject;
+    struct udev_device *udevDevice;
 
-    if ( 0 == storageDescHeader.Size )
+    udevObject = udev_new();
+    if (nullptr == udevObject)
     {
-        return( eErrorCode::Io );   //TODO: replace with more descriptive error
+        throw std::runtime_error("ERROR: Can't create udev \n");
```

Comment:
I saw you throw exceptions so many place, please remember to catch what you thew